lib/src/forms/multi_step_form.dart

```diff
@@ -0,0 +1,157 @@
+import 'package:automated_testing_framework/automated_testing_framework.dart';
+import 'package:automated_testing_framework_plugin_flow_control/automated_testing_framework_plugin_flow_control.dart';
+import 'package:flutter/material.dart';
+import 'package:flutter/scheduler.dart';
+import 'package:static_translations/static_translations.dart';
+
+class MultiStepForm extends TestStepForm {
+  @override
+  Widget buildForm(
+    BuildContext context,
+    Map<String, dynamic> values, {
+    bool minify = false,
+  }) {
+    /// If this line is removed, [values] will still have a reference to the
+    /// original [steps] List, so modifying that here will modify it in the
+    /// [TestStep] without confirmation.
+    values['steps'] = List.from(values['steps'] ?? []);
+    return Column(
+      children: [
+        if (minify != true)
+          buildHelpSection(
+            context,
+            TestFlowControlTranslations.atf_flow_help_multi_step,
+            minify: minify,
+          ),
+        buildValuesSection(
+          context,
+          [
+            buildEditText(
+              context: context,
+              id: 'name',
+              label: TestFlowControlTranslations.atf_flow_form_multi_step_name,
+              values: values,
+            ),
+            _StepsEditor(
+              values: values,
+            )
+          ],
+        ),
+      ],
+    );
+  }
+
+  @override
+  bool get supportsMinified => true;
+
+  @override
+  TranslationEntry get title =>
+      TestFlowControlTranslations.atf_flow_title_multi_step;
+}
+
+class _StepsEditor extends StatefulWidget {
+  _StepsEditor({@required this.values});
+
+  final Map<String, dynamic> values;
+
+  @override
+  _StepsEditorState createState() => _StepsEditorState();
+}
+
+class _StepsEditorState extends State<_StepsEditor> {
+  ScrollController scrollController;
+  List<TestStep> steps = [];
+  Translator translator;
+
+  @override
+  void initState() {
+    super.initState();
+    scrollController = ScrollController();
+    widget.values['steps'] ??= [];
+    steps = (widget.values['steps'] as List)
```

Comment:
I _think_ this can be simplified using:
```
steps = JsonClass.fromDynamicList(
            widget.value['steps'],
            (map) => TestStep.fromDynamic(map))
```

That removes the need for the front cast to a `List` and the final `toList`.